server/logging/factory.go

```diff
@@ -64,6 +64,10 @@ func (f *Factory) CreateIncomingLogger(tokenID string) *AsyncLogger {
 }
 
 func (f *Factory) CreateFailedLogger(destinationName string) *AsyncLogger {
+	if f == nil {
```

Comment:
It is a bad practice to have such checks and return nil. We should have dummy implementation of these structs for overriding behavior, if we don't need a common one.

Reply:
Reverted